src/Platforms/CPU/SIMD/inner_product.hpp

```diff
@@ -211,6 +211,25 @@ inline void transpose(const T* restrict A, size_t m, size_t lda, TO* restrict B,
       B[i * ldb + j] = A[j * lda + i];
 }
 
+/** translation of of A(m,n) to B(m,n) 
+     * @param A starting address, A(m,lda)
+     * @param m number of A rows
+     * @param lda stride of A's row
+     * @param B starting address B(m,ldb)
+     * @param n number of B rows
+     * @param ldb stride of B's row
+     *
+     * Blas-like interface
+     */
+template<typename T, typename TO>
```

Comment:
I can't find a way around needing this when a Matrix that has the bogus extra columns added needs to appear as a Matrix where the cols and rows are assumed to be useful information.